Library/Homebrew/utils/bottles.rb

```diff
@@ -57,6 +57,7 @@ def resolve_version(bottle_file)
     class Bintray
       def self.package(formula_name)
         formula_name.to_s.tr("+", "x")
+        formula_name.to_s.sub!(/(.)@(\d)/, "\\1AT\\2") # Handle foo@1.2 style formulae.
```

Comment:
This isn't doing what it is supposed to do:

```console
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> Utils::Bottles::Bintray.package("gcc")
=> nil
irb(main):002:0> Utils::Bottles::Bintray.package("gcc@6.2")
=> "gccAT6.2"
irb(main):003:0> Utils::Bottles::Bintray.package("crf++")
=> nil
```

You'll need to either append a `sub` (without bang) to the `tr` or introduce a local variable:

```ruby
formula_name.to_s.tr("+", "x").sub(/(.)@(\d)/, "\\1AT\\2")

# or

package_name = formula_name.to_s.dup
package_name.tr!("+", "x")
package_name.sub!(/(.)@(\d)/, "\\1AT\\2")
package_name

# or

package_name = formula_name.to_s.tr("+", "x")
package_name.sub!(/(.)@(\d)/, "\\1AT\\2")
package_name
```

(I'm favoring the 1st variant, that can be split across multiple lines to improve readability, if that's desired.)